### PR TITLE
Clarify metastore filtering and remove unnecessary code

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Comment by losipiuk** [comment]
```
2021-04-30T19:08:01.1159581Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project trino-hive: Compilation failure
2021-04-30T19:08:01.1161964Z [ERROR] /home/runner/work/trino/trino/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:[144,27] [UnusedVariable] The field 'assumeCanonicalPartitionKeys' is never read.
2021-04-30T19:08:01.1163610Z [ERROR]     (see https://errorprone.info/bugpattern/UnusedVariable)
2021-04-30T19:08:01.1164489Z [ERROR]   Did you mean to remove this line or to remove this line?
2021-04-30T19:08:01.1165194Z [ERROR] -> [Help 1]
```

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java [hunk 1]
```diff
@@ -202,6 +203,9 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
             List<String> columnNames,
             TupleDomain<String> partitionKeysFilter)
     {
+        if (isPartitionKeyFilterFalse(partitionKeysFilter)) {
```
Comment (sopel39) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java:206]: I don't think it's needed here. The only caller that does not set `all` (`SemiTransactionalHiveMetastore#doGetPartitionNames`) already handles that case.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java [hunk 2]
```diff
@@ -1063,33 +1062,7 @@ private boolean isValidPartition(Table table, String partitionName)
             List<String> columnNames,
             TupleDomain<String> partitionKeysFilter)
     {
-        if (partitionKeysFilter.isNone()) {
```
Comment (sopel39) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:1066]: Is `getPartitionNamesByFilter` best effort now? I don't think `SemiTransactionalHiveMetastore#doGetPartitionNames` works like that (filters returner results), does it?

**Comment by electrum** [comment]
@losipiuk Thanks, I removed that and found the config in `FileHiveMetastoreConfig` could also be removed.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java (hunk 1 above)
Comment (electrum) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java:206]: Good point, removed

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java (hunk 2 above)
Comment (electrum) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java:1066]: It is best effort. The only caller is `HivePartitionManager#getPartitions` which does the filtering.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/MetastoreUtil.java [hunk 3]
```diff
@@ -324,11 +324,6 @@ public static PrincipalPrivileges buildInitialPrivilegeSet(String tableOwner)
                 ImmutableMultimap.of());
     }
 
-    public static boolean isPartitionKeyFilterFalse(TupleDomain<String> partitionKeysFilter)
-    {
-        return partitionKeysFilter.isNone() || partitionKeysFilter.getDomains().get().values().stream().anyMatch(Domain::isNone);
```
Comment (sopel39) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/MetastoreUtil.java:329]: I don't think `partitionKeysFilter.getDomains().get().values().stream().anyMatch(Domain::isNone)` is needed, see
`io.trino.spi.predicate.TupleDomain#TupleDomain` (it checks `containsNoneDomain`)

**Review by sopel39 — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/MetastoreUtil.java (hunk 3 above)
Comment (electrum) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/MetastoreUtil.java:329]: Ah, you're right, `TupleDomain` guarantees that no domain will be `none`

**Pull request merged**